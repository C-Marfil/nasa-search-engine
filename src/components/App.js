import React, { useEffect, useState } from "react";
import { Route, Routes } from "react-router-dom";
import { throttle } from "lodash";
import Home from "./Home";
import Header from "./Header";
import InspectImage from "./InspectImage";
import SearchResults from "./SearchResults";
import "../styles/app.css";
import BackToTopButton from "./BackToTopButton";

const App = () => {
  const [searchResults, setSearchResults] = useState([]);
  const [level, setLevel] = useState("shallow");

  const handleScroll = throttle(() => {
    if (window.scrollY > 400) {
      setLevel("deep");
    } else if (level === "deep" && window.scrollY <= 200) {
      setLevel("shallow");
    }
  }, 500);

  useEffect(() => {
    window.addEventListener("scroll", handleScroll);
    return () => window.removeEventListener("scroll", handleScroll);
  }, [level, handleScroll]);

  return (
    <div className="app">
      {level === "shallow" && <Header setSearchResults={setSearchResults} />}
      {level === "deep" && <BackToTopButton level={level} />}
      <Routes>
        <Route path="/" element={<Home />} />
        <Route
          path="/:q"
          element={
            <SearchResults
              results={searchResults}
              setLevel={setLevel}
              level={level}
            />
          }
        />
        <Route path="/:q/image-detail" element={<InspectImage />} />
      </Routes>
    </div>
  );
};

export default App;
