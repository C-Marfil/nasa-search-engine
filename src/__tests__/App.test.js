import { React, render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import App from "../components/App";

describe("App", () => {
  it("Snapshot Test", () => {
    const { asFragment } = render(<App />, { wrapper: Router });

    expect(asFragment()).toMatchSnapshot();
  });

  it("tests that elements are rendered", () => {
    const { getByAltText } = render(<App />, { wrapper: Router });
    const image = getByAltText("nasa-logo");

    expect(image).toHaveAttribute(
      "src",
      "https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    );
  });
});
