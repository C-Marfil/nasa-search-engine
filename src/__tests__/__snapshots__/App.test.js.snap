// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      id="header"
    >
      <a
        href="/"
      >
        <img
          alt="nasa-logo"
          class="nasa-logo"
          src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        />
      </a>
      <form
        class="form"
      >
        <input
          class="search-box"
          placeholder="Search here"
          type="text"
        />
        <button
          class="button"
          type="submit"
        >
          GO
        </button>
      </form>
    </div>
    <div
      class="sc-bwzfXH lhcaPW moving-words"
      direction="alternate-reverse"
      type="pulse"
    >
      Welcome, use the Search Bar above to find any pictures held in the public NASA archives.
    </div>
  </div>
</DocumentFragment>
`;
